### PR TITLE
Do not require CDTs for qt6

### PR DESCRIPTION
**Pull request by hmaarrfk**
Manually ran a diff on the logs between qt6 branch and PR of:
- [x] linux_64
    - No differences visually seen in the `Configure summary`
- [x] linux_aarch64
    - I notice the the native build seems many more opengl options enabled.
    - However, the cross compiled build has the same options enabled as before.
- [x] osx x86_64
    - It sees that `QDoc` and `Clang-based lupdate parser` are now enabled, they were not before.
- [x] osx arm
   - No observed differences in the `Configure Summary`

Logs with timestamps removed so you can see a "vimdiff"
[compare_qt_logs.zip](https://github.com/conda-forge/qt-main-feedstock/files/11138304/compare_qt_logs.zip)

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/build.sh [hunk 1]
```diff
@@ -61,6 +72,7 @@ cmake -LAH -G "Ninja" ${CMAKE_ARGS} \
   -DFEATURE_openssl_linked=ON \
   -DFEATURE_designer=OFF \
   -DFEATURE_linguist=OFF \
+  ${FEATURE_DEFINES} \
```
Comment (hmaarrfk) [inline recipe/build.sh:75]:
```suggestion
```

**Comment by hmaarrfk** [comment]
Bootstrapping strategy is discussed (lightly) https://github.com/conda-forge/pyside2-feedstock/pull/136/files#r997301030

File: recipe/build.sh [hunk 2]
```diff
@@ -27,6 +28,7 @@ then
       -DFEATURE_qml_animation=OFF \
       -DQT_BUILD_SUBMODULES="qtbase;qtdeclarative;qtshadertools;qttools" \
       -DCMAKE_RANLIB=$BUILD_PREFIX/bin/${CONDA_TOOLCHAIN_BUILD}-ranlib \
+      -DFEATURE_gui=OFF \
```
Comment (hmaarrfk) [inline recipe/build.sh:31]:
```suggestion
```

**Comment by hmaarrfk** [comment]
A few x libraries that may be missing:

- `libxss`
- `X11_Xxf86misc`

File: recipe/yum_requirements.txt [hunk 3]
```diff
@@ -1,23 +1,5 @@
-alsa-lib
-libXScrnSaver
-libXtst
-libXi
-libXcursor
-libXcomposite
 mesa-libGL
+mesa-libEGL-devel
```
Comment (hmaarrfk) [inline recipe/yum_requirements.txt:2]:
```suggestion
mesa-libGL-devel
mesa-libEGL-devel
```

**Comment by hmaarrfk** [comment]
todo: consider specifying more features on linux.

```
FEATURE_DEFINES=""
if [[ "${target_platform}" == "linux-64" ]]; then
    # Do we even want EGL or EGLFS at all?? they were never enabled for
    # any linux platform
    FEATURE_DEFINES="${FEATURE_DEFINES} -DFEATURE_egl=ON"
    FEATURE_DEFINES="${FEATURE_DEFINES} -DFEATURE_eglfs=ON"
fi
```

File: recipe/build.sh [hunk 4]
```diff
@@ -38,6 +53,15 @@ then
   CMAKE_ARGS="${CMAKE_ARGS} -DQT_HOST_PATH=${BUILD_PREFIX} -DQT_FORCE_BUILD_TOOLS=ON -DBUILD_WITH_PCH=OFF"
 fi
 
+FEATURE_DEFINES=""
+if [[ "${target_platform}" == "linux-64" ]]; then
```
Comment (hmaarrfk) [inline recipe/build.sh:57]:
```suggestion
if [[ "${target_platform}" == "linux_64" ]]; then
```

**Comment by hmaarrfk** [comment]
@jschueller what do you think of this approach. I believe it got through locally.

I still have to manually make sure I haven't accidentally removed some features but I'm hopeful

**Comment by jschueller** [comment]
this is fine with me

**Comment by hmaarrfk** [comment]
osx arm seems to timeout from time to time

**Pull request merged**